File: TechDeck.Identity/Models/UserAuthBase.cs
namespace TechDeck.Identity.Models
{
    public record UserClaim(
        Guid ClaimId,
        Guid UserId,
        string ClaimType,
        string ClaimValue);

    public record UserAuthBase(
        int UserId,
        string UserName,
        string BearerToken,
        bool IsAuthenticated,
        List<UserClaim> Claims);
}

File: TechDeck.Web/src/app/security/user-auth-base.ts
export class UserAuthBase {
  userId: number = 0;
  userName: string = "";
  bearerToken: string = "";
  isAuthenticated: boolean = false;
  claims: AppUserClaim[] = [];
  name: string = "";

  init(): void {
    this.userName = "";
    this.bearerToken = "";
    this.isAuthenticated = false;
    this.claims = [];
    this.name = "";
  }
}

export class AppUserClaim  {
  claimId: string = "";
  userId: string = "";
  claimType: string = "";
  claimValue: string = "";
}
